src/Service/VendorService/TheMovieDatabase/TheMovieDatabaseApiService.php

```diff
@@ -0,0 +1,212 @@
+    }
+
+    /**
+     * Search in the movie database for a poster url by title, year and director.
+     *
+     * @param string $title
+     *   The title of the item
+     * @param string $originalYear
+     *   The release year of the item
+     * @param string|null $director
+     *   The director of the movie
+     *
+     * @return string
+     *   The poster url or null
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function searchPosterUrl(string $title = null, string $originalYear = null, string $director = null): ?string
+    {
+        $posterUrl = null;
+
+        // Bail out if the required information is not supplied.
+        if (null === $title || null === $originalYear || null === $director) {
+            return $posterUrl;
+        }
+
+        try {
+            $query = [
+                'query' => [
+                    'query' => $title,
+                    'year' => $originalYear,
+                    'api_key' => $this->apiKey,
+                    'page' => '1',
+                    'include_adult' => 'false',
+                    'language' => 'da_DK',
+                ],
+            ];
+
+            $responseData = $this->sendRequest(self::SEARCH_URL, $query);
+
+            $result = $this->getResultFromSet($responseData->results, $title, $director);
+
+            if ($result) {
+                $posterUrl = $this->getPosterUrl($result);
+            }
+        } catch (\Exception $e) {
+            $posterUrl = null;
+        }
+
+        return $posterUrl;
+    }
+
+    /**
+     * Get the first match in the result set.
+     *
+     * @param array $results
+     *   Array of search results
+     * @param string $title
+     *   The title of the item
+     * @param string $director
+     *   The director of the item
+     *
+     * @return \stdClass|null
+     *   The matching result or null
+     */
+    private function getResultFromSet(array $results, string $title, string $director): ?\stdClass
+    {
+        $chosenResult = null;
+
+        foreach ($results as $result) {
+            // Validate title againt result->title or result->original_title.
+            if (strtolower($result->title) === strtolower($title) || strtolower($result->original_title) === strtolower($title)) {
+                // Validate director.
+                try {
+                    // https://developers.themoviedb.org/3/movies/get-movie-credits
+                    $queryUrl = 'https://api.themoviedb.org/3/movie/'.$result->id.'/credits';
+                    $responseData = $this->sendRequest($queryUrl);
+
+                    $directors = array_reduce($responseData->crew, function ($carry, $item) {
+                        if ('Director' === $item->job) {
+                            $carry[] = $item->name;
+                        }
+
+                        return $carry;
+                    }, []);
+
+                    if (in_array($director, $directors)) {
```

Comment:
Same here is the UTF8 chars the same or would a normalizer mb_string to lower help?

Reply:
change to mb_strtolower